sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs

```diff
@@ -93,6 +93,44 @@ internal static void CastAndRemoveAnnotations(this IReadOnlyList<IDictionary<str
             }
         }
 
+        /// <summary>
+        /// Cleans a Dictionary of its Odata type annotations, while using them to cast its entities accordingly.
+        /// </summary>
+        internal static void CastAndRemoveAnnotations(this IDictionary<string, object> entity)
+        {
+            var typeAnnotationsWithKeys = new Dictionary<string, (string typeAnnotation, string annotationKey)>();
+            var spanOdataSuffix = TableConstants.Odata.OdataTypeString.AsSpan();
+
+            typeAnnotationsWithKeys.Clear();
+
+            foreach (var propertyName in entity.Keys)
```

Comment:
There is a bit of duplication with other CastAndRemoveAnnotations but might not matter yet.

Reply:
Yes, it is mostly dupe, but it got tricky to convert into a reusable method so I left it this way for now.